liquibase-core/src/main/java/liquibase/database/AbstractJdbcDatabase.java

```diff
@@ -763,7 +763,10 @@ public void dropDatabaseObjects(final CatalogAndSchema schemaToDrop) throws Liqu
             }
 
 	        final long changeSetStarted = System.currentTimeMillis();
-	        DiffResult diffResult = DiffGeneratorFactory.getInstance().compare(new EmptyDatabaseSnapshot(this), snapshot, new CompareControl(snapshot.getSnapshotControl().getTypesToInclude()));
+            CompareControl compareControl = new CompareControl(
```

Comment:
diffResult via clearDatabase and DropAllCommand was empty.

Adding schemaComparison seems legid based on schemaToDrop parameter.